app/src/main/java/com/asfoundation/wallet/promo_code/bottom_sheet/PromoCodeBottomSheetFragment.kt

```diff
@@ -186,8 +179,15 @@ class PromoCodeBottomSheetFragment : DaggerBottomSheetDialogFragment(),
     views.promoCodeBottomSheetSuccessAnimation.playAnimation()
     views.promoCodeBottomSheetSuccessTitle.visibility = View.VISIBLE
     views.promoCodeBottomSheetSuccessSubtitle.visibility = View.VISIBLE
-    views.promoCodeBottomSheetSuccessSubtitle.text =
-        this.getString(R.string.promo_code_success_body, bonus.toString())
+    if (promoCode.appName != null) {
+      views.promoCodeBottomSheetSuccessSubtitle.text =
+          this.getString(R.string.promo_code_success_body_specific_app, promoCode.bonus.toString(),
+              promoCode.appName)
+    } else {
+      views.promoCodeBottomSheetSuccessSubtitle.text =
+          this.getString(R.string.promo_code_success_body, promoCode.bonus.toString())
+    }
```

Comment:
I don't know if you agree with this, but I believe putting the setText before the if seems cleaner and more in line with the lines above this.
![image](https://user-images.githubusercontent.com/13988658/142501052-cad65f2e-1ab9-4ade-bd2e-d8af739046d3.png)

Reply:
Normally, I'd agree for simple if / else statements, but this one is so big that I don't think there's value in it. I guess we could discuss this in code style meeting.